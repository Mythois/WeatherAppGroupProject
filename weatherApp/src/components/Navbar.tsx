import './styles/Navbar.css'
import { Link } from "react-router-dom"
import React, {useState} from 'react'

export const Navbar = () => {
    const [menuOpen, setmenuOpen] = useState(false)
    return (<nav>
        <div className='menu' onClick={() =>{setmenuOpen(!menuOpen)}}>
            <span></span>
            <span></span>
            <span></span>
        </div>
        <div className='logoContainer'><h2>WebDevWeather</h2></div>
        <ul className={menuOpen ? "open" :""}>
            <li className="navigationElement"><Link to="/Home">Home</Link></li>
            <li className="navigationElement"><Link to="/Favourites">Favourites</Link></li>
        </ul>
    </nav>)
}