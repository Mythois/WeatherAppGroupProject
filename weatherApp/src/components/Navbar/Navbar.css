nav{
    width:100%;
    display: flex;
    justify-content: flex-end;
    flex-direction: row;
    gap:70%;
    color: white;
    background-color: var(--compBg);
}

ul{
    list-style-type: none;
    margin-right:5%;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    gap: 30%;
}

a{
    text-decoration: none;
    color: black;
}

a:visited{
    color: black;
}

.navigationElement{
    display: inline-block;
    position:relative;
    padding-bottom: 0.25rem;  
  }
  
  @keyframes fade-in {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }

  .navigationElement::before{
    content:"";
    position: absolute;
    left:0px;
    bottom: 0px;
    width:0%;
    height:2px;
    background-color: green;
    transition: width 0.25s ease-out;
  }
  
  .navigationElement:hover::before{
    width: 100%;
  }

.menu{
    display: none;
    position: absolute;
    top: 1.5rem;
    right:1rem;
    flex-direction: column;
    justify-content: space-between;
    width: 2.25rem;
    height: 2rem;
}

nav .menu span{
    height:0.4rem;
    width: 100%;
    background-color: black;
    border-radius:5px;
}

.logoContainer{
    color:black;
}



@media(max-width: 480px){
    nav{
        flex-direction: column;
        align-items: center;
    }
    .menu{
        display:flex;
    }

    ul{
        display: none;
        flex-direction: column;
        width:100%;
    }

    ul.open{
        display: flex;
    }
    
    .navigationElement{
        width:100%;
        text-align: center;
    }

    a{
        margin: 0.2rem 0.5rem;

    }
}