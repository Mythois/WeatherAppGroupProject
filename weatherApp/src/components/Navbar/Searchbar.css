.searchbar-container {
    position: relative;
    display: flex;
    justify-content: space-between;
    align-items: center;
}
  
.searchbar-container input {
    width: 100%;
    height: 100%;
    padding: 0;
    margin: 0;
    border: none;
    border-bottom: 1px solid #ccc;
    font-size: 16px;
}
  
.searchbar-container button {
    margin-left: 10px; 
    padding: 5px 10px; 
    font-size: 16px; 
}
  
.suggestions-list {
    position: absolute;
    left: 0;
    top: calc(100% + 5px);
    list-style: none;
    padding: 0;
    margin: 0;
    max-height: 150px;
    overflow-y: auto;
}
  
.suggestion-item {
    cursor: pointer;
}
  