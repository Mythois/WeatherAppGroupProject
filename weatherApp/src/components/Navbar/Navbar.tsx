import React, { useState } from 'react';
import './Navbar.css';

const Navbar = () => {
  const [menuOpen, setMenuOpen] = useState(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  return (
    <nav>
      <div className={`menu ${menuOpen ? 'open' : ''}`} onClick={toggleMenu}>
        <div className="bar1"></div>
        <div className="bar2"></div>
        <div className="bar3"></div>
      </div>
      <div className='logoContainer'><h2>WebDevWeather</h2></div>
      <ul className={menuOpen ? "open" : ""}>
        <li className="navigationElement"><a href="/Home">Home</a></li>
        <li className="navigationElement"><a href="/Favourites">Favourites</a></li>
      </ul>
    </nav>
  );
}

export default Navbar;
