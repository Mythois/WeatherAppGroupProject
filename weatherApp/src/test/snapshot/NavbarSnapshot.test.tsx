import { render } from '@testing-library/react'
import Navbar from '../../components/Navbar/Navbar'

describe('Navbar component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Navbar />)
    expect(asFragment()).toMatchSnapshot()
  })
})
