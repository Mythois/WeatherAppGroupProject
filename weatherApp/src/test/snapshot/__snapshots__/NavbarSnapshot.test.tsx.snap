// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > matches snapshot > src/test/snapshot/__snapshots__/NavbarSnapshot.test.tsx.snap 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="menu "
    >
      <div
        class="bar1"
      />
      <div
        class="bar2"
      />
      <div
        class="bar3"
      />
    </div>
    <div
      class="logoContainer"
    >
      <h2>
        <a
          class="linkstyle"
          href="/project1/"
        >
          WebDevWeather
        </a>
      </h2>
    </div>
    <ul
      class=""
    >
      <li
        class="navigationElement"
      >
        <a
          class="linkstyle"
          href="/project1/Home"
        >
          Home
        </a>
      </li>
      <li
        class="navigationElement"
      >
        <a
          class="linkstyle"
          href="/project1/Favorites"
        >
          Favourites
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
